### PR TITLE
CP-11136 Virtual Hardware Platform Version (creedence-bugfix branch)

### PR DESCRIPTION
**Pull request by thomassa**
For https://xapi-project.github.io/xen-api/futures/virt-hw-platform-vn/virt-hw-platform-vn.html

(Note for future: when cherrypicking these changes from creedence-bugfix to creedence-sp we must check whether we need to increase the datamodel (and cream) schema minor version again.)

### CODE REVIEW

**Comment by thomassa** [comment]
Oops: don't pull this until I've updated the datamodel version for the new fields.

**Comment by thomassa** [comment]
This is based on selected parts of the code-changes from #2138.
@djs55 has reviewed most of it (bar name-change and version-bump, but with unrelated bits that have been excluded from this pullreq on purpose) on #2123 at https://github.com/xapi-project/xen-api/pull/#2123#issuecomment-72895183
> OK, apart from the `virt-hw-vn` name, think this is great :+1: 

but it makes sense to review this pullreq in case of any mistakes in picking out the relevant changes etc.

**Comment by thomassa** [comment]
Oops: when I incremented the datamodel schema minor version to 73 I should have done the same for  `cream_release_schema_minor_vsn`.
I'll fix that now.

File: ocaml/xapi/xapi_globs.ml [hunk 1]
```diff
@@ -838,3 +838,24 @@ let read_external_config () =
...
+	   introduced. This Virtual Hardware Platform might not differ
+	   significantly from the immediately preceding version zero, but
+	   it seems prudent to introduce a way to differentiate it from
+	   the whole history of older host versions. *)
+	1L;
+	(* We anticipate that version two will be the first virtual
+	   hardware platform version to offer the option of an emulated
+	   PCI device used to trigger a guest to install or upgrade its PV
+	   tools (originally introduced to exploit the Windows Update
+	   system). *)
```
Comment (djs55) [inline ocaml/xapi/xapi_globs.ml]: Good explanation, very helpful!

**Comment by djs55** [comment]
This looks good to me.

**Comment by thomassa** [comment]
Merging in the light of the code-review above.

**Pull request merged**